From 8930658f50b9be61ad8e0c8c9189be1e27d0bcf4 Mon Sep 17 00:00:00 2001
From: Theo GOUREAU <theo.goureau-ext@st.com>
Date: Tue, 5 Nov 2024 17:48:50 +0100
Subject: [PATCH 1356/1695] plat-stm32mp2: disable stm32_debug_suspend() when
 debug is disabled

Disable stm32_debug_suspend() when BSEC_DENR_DBGSWEN == 0, i.e.,
when the debug is disabled.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 64b70e8b15fb586d94d7a52e130c096c0d6f6aae plat-stm32mp2: disable stm32_debug_suspend() when debug is disabled]
Change-Id: Id120c491d0563b5530dc4172a4b54f788d5e12c4
Signed-off-by: Theo GOUREAU <theo.goureau-ext@st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/417288
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
Reviewed-by: Etienne CARRIERE <etienne.carriere@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
---
 core/arch/arm/plat-stm32mp2/main.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/core/arch/arm/plat-stm32mp2/main.c b/core/arch/arm/plat-stm32mp2/main.c
index e9448f53a..84d9d08da 100644
--- a/core/arch/arm/plat-stm32mp2/main.c
+++ b/core/arch/arm/plat-stm32mp2/main.c
@@ -509,6 +509,9 @@ void stm32_debug_suspend(unsigned long a0)
 	uint32_t dbgmcu_cr = U(0);
 	static bool info_displayed;
 
+	if (!stm32_bsec_self_hosted_debug_is_enabled())
+		return;
+
 	if (clk_enable(dbgmcu_clk))
 		return;
 
-- 
2.47.2

